src/Library.php

```diff
@@ -183,4 +199,54 @@ public static function delete_expired_draft_orders() {
 			"
 		);
 	}
+
+	/**
+	 * Attempt to process a payment for the checkout API if no payment methods support the
+	 * woocommerce_rest_checkout_process_payment_with_context action.
+	 *
+	 * @param PaymentContext $context Holds context for the payment.
+	 * @param PaymentResult  $result  Result of the payment.
+	 */
+	public static function process_legacy_payment( PaymentContext $context, PaymentResult &$result ) {
+		if ( $result->status ) {
+			return;
+		}
+
+		// phpcs:ignore WordPress.Security.NonceVerification
+		$post_data = $_POST;
```

Comment:
Currently payment methods in Woo Core probably rely on the nonce verification that happens in the `WC()->checkout->process_checkout()` method right?  Do we not care about that because this is covered by nonce validation in our REST endpoint?

Reply:
Correct; we have our own nonce protection and gateways are not responsible for this either.